singularity-opac/photons/mean_photon_variant.hpp

```diff
@@ -0,0 +1,99 @@
+// ======================================================================
+// © 2022. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+// ======================================================================
+
+#ifndef SINGULARITY_OPAC_PHOTONS_MEAN_PHOTON_VARIANT_
+#define SINGULARITY_OPAC_PHOTONS_MEAN_PHOTON_VARIANT_
+
+#include <utility>
+
+#include <ports-of-call/portability.hpp>
+#include <singularity-opac/base/opac_error.hpp>
+#include <singularity-opac/base/radiation_types.hpp>
+#include <singularity-opac/photons/photon_variant.hpp>
+#include <variant/include/mpark/variant.hpp>
+
+namespace singularity {
+namespace photons {
+namespace impl {
+
+template <typename... Opacs>
+class MeanVariant {
```

Comment:
Can you remind me why mean opacities needed a separate API? Also will we ever want anything other than the tables? I.e., is there a reason to make a whole variant?

Reply:
It's only for scale-free vs CGS forms (whether we're using the trivial thermal distribution function where all the constants are `1`)

Reply:
ooh I see ok.